frontend/schemaCMS/src/shared/components/blockElement/blockElement.component.js

```diff
@@ -23,19 +23,21 @@ export const BlockElement = props => {
           <IconContainer>
             <MinusIcon />
           </IconContainer>
-          <TextInput
-            name={elementPath}
-            placeholder={intl.formatMessage(messages.elementNamePlaceholder)}
-            onChange={handleChange}
-            autoWidth
-            fullWidth
-            value={element.name}
-            customInputStyles={customInputStyles}
-            {...restFormikProps}
-          />
-          <IconContainer>
-            <EditIcon />
-          </IconContainer>
+          {isAdmin ? (
+            <TextInput
+              name={elementPath}
+              placeholder={intl.formatMessage(messages.elementNamePlaceholder)}
+              onChange={handleChange}
+              autoWidth
+              fullWidth
+              value={element.name}
+              customInputStyles={customInputStyles}
+              {...restFormikProps}
+            />
+          ) : (
+            element.name
+          )}
```

Comment:
can we have `renderWhenAdmin`  in this component at least?

Reply:
Or even `renderWhenTrueOtherwise`

Reply:
Oh, I forget to change here also. Thanks for cathing 👍 .

Reply:
Fixed :)